[weiyanminoo] iP

weiyanminoo opened the pull request:
# Baymax

> If knowledge is power, learning is your superpower - [Jim Kwik](https://iamfearlesssoul.com/jim-kwik-quotes/)

Baymax frees your mind of having to remember things to do, giving you more time to learn! It's

- text-based
- simple to pick-up
- ~~fast~~ incredibly fast to use

## What you need to do:

1. Download it from [here](https://github.com/weiyanminoo/ip/releases/tag/A-Jar)
2. Double click on it
3. Add your tasks
4. Let Baymax manage your tasks for you 😄 

And BEST of all, its **FREE**!

## What Baymax can do for you:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Sending reminders (coming soon)

What are you waiting for? Download Baymax today 😎

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename, Greet, Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark As Done

Commit:
Todos, Deadlines, Events

Commit:
Automated Text UI Testing

Commit:
Handle Errors

Commit:
Delete

Commit:
Enum

Commit:
Add Storage.java

Commit:
Add Storage.java

Commit:
Update for storage and task classes

Commit:
Add improvements to Storage class

Commit:
Merge branch 'branch-Level-7'

Commit:
Debug feature for saving tasks

Commit:
Update deadline.java to understand dates

Commit:
Update event.java to understand dates

Commit:
Update Baymax.java to work with new date format

Commit:
Merge branch 'branch-Level-8'

Commit:
Debug baymax.java for time formatting for events

Commit:
Add UI.java

Commit:
Add TaskList.java

Commit:
Add Parser.java

Commit:
Update Baymax.java to work with new classes

Commit:
Merge branch 'branch-oop'

Commit:
Debug saving and loading tasks.txt

Commit:
Add packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add two JUnit tests

Commit:
Update build.gradle

Commit:
Add header comments

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Update TaskList.java for delete method

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Add 'find task' feature

Commit:
Resolve merge conflicts (branch 9)

Commit:
Add line indentation

AndreChia commented on src/main/java/baymax/storage/Storage.java:
```diff
@@ -0,0 +1,108 @@
+            String line = scanner.nextLine();
+            String[] parts = line.split(" \\| ");
+
+            // Catch corrupted files by detecting invalid content format
```
Instead of using IllegalArgumentException, using a customized exception would improve error clarity. Maybe InvalidTaskFormatException could provide more descriptive debugging information?

AndreChia commented on src/main/java/baymax/task/Event.java:
```diff
@@ -0,0 +1,86 @@
+     *
+     * @return The formatted Event task string.
+     */
+    @Override
```
Using String.format(...) would improve readability. Concatenating multiple .format() calls might reduce clarity, and a formatted string might make it easier to adjust in the future.

AndreChia commented on src/main/java/baymax/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        String[] words = input.split(" ", 2);
+        String command = words[0];
+
+        switch (command) {
```
Returning a status flag instead of calling System.exit(0) directly would be better. Since we are implementing a GUI, handling termination at the Parser level removes control from the main application, which can cause issues.

AndreChia replied:
Would it be beneficial to use an enum for command types instead of relying on raw string comparisons here?

AndreChia commented on src/main/java/baymax/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+                taskList.markTask(parseIndex(words), false, ui);
+                break;
+            case "delete":
+                taskList.deleteTask(parseIndex(words), ui);
```
Would it be beneficial to use an enum for command types instead of relying on raw string comparisons here?

AndreChia reviewed (commented):
Overall, LGTM! 👍 Great job with maintaining clear and consistent naming conventions, keep up the excellent work!

AndreChia commented on src/main/java/baymax/storage/Storage.java:
```diff
@@ -0,0 +1,108 @@
+            // Catch corrupted files by detecting invalid content format
+            try {
+                if (parts.length < 3) {
+                    throw new IllegalArgumentException("Invalid task format: " + line);
```
Instead of using IllegalArgumentException, using a customized exception would improve error clarity. Maybe InvalidTaskFormatException could provide more descriptive debugging information?

cscms03 replied:
Having custom exception class could be appropriate, and maybe make another package that groups all of them.

AndreChia commented on src/main/java/baymax/task/Event.java:
```diff
@@ -0,0 +1,86 @@
+     */
+    @Override
+    public String toFileFormat() {
+        return "E | " + (isDone ? "1" : "0") + " | " + description + " | " + date + " | " + from + " | " + to;
```
Using String.format(...) would improve readability. Concatenating multiple .format() calls might reduce clarity, and a formatted string might make it easier to adjust in the future.

AndreChia commented on src/main/java/baymax/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        switch (command) {
+            case "bye":
+                ui.exitMessage();
+                System.exit(0);
```
Returning a status flag instead of calling System.exit(0) directly would be better. Since we are implementing a GUI, handling termination at the Parser level removes control from the main application, which can cause issues.

jingchun19 commented on src/main/java/baymax/task/TaskType.java:
```diff
@@ -0,0 +1,5 @@
+package baymax.task;
+
+public enum TaskType {
+    TODO, DEADLINE, EVENT
```
Great use of enum

jingchun19 commented on src/main/java/baymax/tasklist/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+
+    private ArrayList<Task> tasks;
+    private Storage storage;
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter TIME_FORMAT = DateTimeFormatter.ofPattern("HHmm");
```
Using singular d and M would allow the DataTime formatter to take in both single and double digit integers

jingchun19 commented on src/main/java/baymax/ui/UI.java:
Amazing use of UI class

jingchun19 commented on src/main/java/baymax/tasklist/TaskList.java:
```diff
@@ -0,0 +1,265 @@
+     * @param ui The UI instance for user interaction.
+     * @throws BaymaxException If the input format is invalid.
+     */
+    public void addDeadline(String input, UI ui) throws BaymaxException {
```
Good work on trying to catch any error that the user might face!

jingchun19 reviewed (commented):
Overall quite a clean code base!

cscms03 commented on src/main/java/baymax/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        String command = words[0];
+
+        switch (command) {
+            case "bye":
```
I think use of ENUM could be an option here. Also can maybe add toLowerCase() method for more flexibility.

cscms03 commented on src/main/java/baymax/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+     * @param words The split user input containing the command and index.
+     * @return The parsed index.
+     * @throws BaymaxException If the index is missing or invalid.
+     */
```
great JavaDoc with complying conventions

cscms03 commented on src/main/java/baymax/storage/Storage.java:
```diff
@@ -0,0 +1,108 @@
+        }
+        writer.close();
+        // Used for debugging
+        // System.out.println("Tasks saved to: " + new File(FILE_PATH).getAbsolutePath());
```
commented code lines could be removed for final push.

cscms03 reviewed (commented):
Overall, good naming conventions and relatively neat code to read. I think it quite follows coding standard

Commit:
Integrate JavaFX

Commit:
Merge branch 'branch-Level-10'

weiyanminoo committed:
Update TaskListTest.java

This test class could not compile after I made some changes to TaskList.java due to the past week's requirements. Some edge cases, such as invalid inputs and boundary conditions, are also not handled well previously.

This commit adds unit tests for adding tasks, deleting tasks and marking tests to improve code reliability and prevent regression.

weiyanminoo committed:
Add Assertions for Task.java

Task.java lacked assertions to validate input data, which could lead to unexpected errors and inconsistencies.

Let's
- add assertions in the constructor to verify that descriptions are non-null and non-empty.
- add assertions in fromFileFormat() to validate input data before processing.

These changes enhance the reliability of the program

weiyanminoo committed:
Add assertions for Deadline.java and Event.java

Deadline.java and Event.java lacked assertions to validate input data, which could lead to unexpected errors and inconsistencies.

Let's,
- add assertions in constructors to verify that descriptions and date/time values are non-null and correctly formatted.
- ensure that deadlines and event durations are valid upon creation.

These changes enhance the reliability of the program.

weiyanminoo committed:
Update switch statements

weiyanminoo committed:
Add Assertions for Parser.java and Storage.java

weiyanminoo committed:
Add assertions to TaskList.java

weiyanminoo committed:
Merge pull request #2 from weiyanminoo/branch-A-Assertions

Add Assertions to various classes

weiyanminoo committed:
Sync local master with remote master branch

weiyanminoo committed:
Merge branch 'master' into branch-A-CodeQuality

Resolve conflicts in Task.java

weiyanminoo committed:
Check for code quality

weiyanminoo committed:
Merge pull request #3 from weiyanminoo/branch-A-CodeQuality

Check for code quality issues

weiyanminoo committed:
Merge branch 'master' of https://github.com/weiyanminoo/ip

weiyanminoo committed:
Update 'Find Task' feature

weiyanminoo committed:
Add 'Undo' feature

weiyanminoo committed:
Add 'undo' feature for markTask()

weiyanminoo committed:
Merge branch 'branch-BCD-Extension'

weiyanminoo committed:
Remove UI.java class

weiyanminoo committed:
Add 'Baymax' Label for GUI

weiyanminoo committed:
Add Error Handling for Deadline and Event in TaskList.java

weiyanminoo committed:
Improve GUI

Make the UI for the bot and user differnet

Add emphasis on the error messages

weiyanminoo committed:
Provide final edits

weiyanminoo committed:
Add README

weiyanminoo committed:
Update README

weiyanminoo committed:
Modify GUI and Debug TaskList.java

weiyanminoo committed:
Add jar file link

weiyanminoo committed:
Modify code after smoke test comments

weiyanminoo committed:
Merge branch 'master' of https://github.com/weiyanminoo/ip

weiyanminoo committed:
Add last modifications

weiyanminoo committed:
Add final updates to README

weiyanminoo committed:
Add minor modification to README
